src/shared/dissect-image.h

```diff
@@ -106,17 +125,107 @@ static inline PartitionDesignator PARTITION_VERITY_SIG_OF(PartitionDesignator p)
         case PARTITION_ROOT_SECONDARY:
                 return PARTITION_ROOT_SECONDARY_VERITY_SIG;
 
+        case PARTITION_ROOT_OTHER:
+                return PARTITION_ROOT_OTHER_VERITY_SIG;
+
         case PARTITION_USR:
                 return PARTITION_USR_VERITY_SIG;
 
         case PARTITION_USR_SECONDARY:
                 return PARTITION_USR_SECONDARY_VERITY_SIG;
 
+        case PARTITION_USR_OTHER:
+                return PARTITION_USR_OTHER_VERITY_SIG;
+
         default:
                 return _PARTITION_DESIGNATOR_INVALID;
         }
 }
 
+static inline PartitionDesignator PARTITION_ROOT_OF_ARCH(Architecture arch) {
+        switch (arch) {
+
+        case native_architecture():
+                return PARTITION_ROOT;
+
+        case ARCHITECTURE_SECONDARY:
+                return PARTITION_ROOT_SECONDARY;
+
+        default:
+                return PARTITION_ROOT_OTHER;
+        }
+}
+
+static inline PartitionDesignator PARTITION_ROOT_VERITY_OF_ARCH(Architecture arch) {
+        switch (arch) {
+
+        case native_architecture():
+                return PARTITION_ROOT_VERITY;
+
+        case ARCHITECTURE_SECONDARY:
+                return PARTITION_ROOT_SECONDARY_VERITY;
+
+        default:
+                return PARTITION_ROOT_OTHER_VERITY;
+        }
+}
+
+static inline PartitionDesignator PARTITION_ROOT_VERITY_SIG_OF_ARCH(Architecture arch) {
```

Comment:
this is redundant, no? because `PARTITION_VERITY_SIG_OF(PARTITION_ROOT_OF_ARCH(arch))` would return the same, no?

In order to minimize the databases we keep, maybe either use that everywhere or at least define PARTITION_ROOT_VERITY_SIG_OF_ARCH() as that expression?